### PR TITLE
[linstor] Update drbd to 9.2.4

### PR DESCRIPTION
**Pull request by duckhawk**
## Description

Update drbd to 9.2.4. Added disruptive step for updating.

Co-developed-by: Andrei Kvapil (@kvaps)

## Why do we need it, and what problem does it solve?

This is patch release, that fixes multiple errors in drbd. Also, it adds check of drbd version on nodes, and step with node reboot, if it necessary.

## What is the expected result?

Fix multiple issues in drbd. Actual drbd version in clusters.
- https://github.com/LINBIT/linstor-server/issues/329
- https://github.com/LINBIT/linstor-server/issues/321
- Resolves possible deadlocks due to network failure

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: linstor
type: chore
summary: Update drbd to 9.2.4
impact: | 
  This update requires disruption, so all nodes with linstor will be restarted.
  * If NodeGroup spec.disruptions.approvalMode is set to Manual, you will recieve alert NodeRequiresDisruptionApprovalForUpdate.
  * If NodeGroup spec.disruptions.approvalMode is set to Automatic, nodes will be drained and restartred one by one.
impact_level: high
```

### CODE REVIEW

File: modules/041-linstor/images/drbd-driver-loader/Dockerfile
Comment (kvaps) [inline modules/041-linstor/images/drbd-driver-loader/Dockerfile:1]: This change will remove SELinux support from our setups. The upstream introduced new environment variable to control this behavior. Please take a look https://github.com/LINBIT/drbd/commit/8ecbec8e4bc533e0733e5b595b8f9b671c11bd04

**Review by kvaps — CHANGES_REQUESTED** [review]

File: modules/041-linstor/images/drbd-driver-loader/Dockerfile
Comment (duckhawk) [reply modules/041-linstor/images/drbd-driver-loader/Dockerfile:1]: Thanks, noticed this too late

**Review by kvaps — DISMISSED** [review]
lgtm

**Review by kvaps — COMMENTED** [review]
Since it include changes from #4788 could you add

```
Co-authored-by: Andrei Kvapil <kvapss@gmail.com>
```

to the commit message please

File: modules/041-linstor/hack/update.sh [hunk 1]
```diff
@@ -50,5 +50,7 @@ while read name repo; do
 done < <(echo "$gitrepos" | sed -n 's|_GITREPO=| |p')
 
 echo "Applying changes:"
-set -x
-sed -e "$sed_regex" -i $targets
+(set -x; sed -e "$sed_regex" -i $targets)
+if [ -n "$drbd_alert_version" ]; then
```
Comment (apolovov) [inline modules/041-linstor/hack/update.sh:54]: What alert? The pr don't provide any alerts.

**Review by apolovov — CHANGES_REQUESTED** [review]

File: modules/041-linstor/images/spaas/Dockerfile [hunk 2]
```diff
@@ -29,6 +29,7 @@ RUN git clone ${DRBD_GITREPO} /drbd \
 #  && wget ${DRBD_PKG}
 
 FROM $BASE_UBUNTU
+ARG DRBD_VERSION=9.2.4
```
Comment (kvaps) [inline modules/041-linstor/images/spaas/Dockerfile:32]:
```suggestion
```

File: modules/041-linstor/images/spaas/Dockerfile [hunk 3]
```diff
@@ -42,4 +43,5 @@ RUN apt-get update \
  && ln -sf /proc/mounts /etc/mtab
 COPY --from=builder /spaas /
 COPY --from=builder /drbd/drbd-*.tar.gz /var/cache/spaas/tarballs/
+RUN echo ${DRBD_VERSION} > /drbd_version
```
[the remark this answers is not in the document]
Comment (kvaps) [reply modules/041-linstor/images/spaas/Dockerfile:46]:
```suggestion
```

**Review by apolovov — DISMISSED** [review]

**Review by nabokihms — APPROVED** [review]

File: modules/041-linstor/images/drbd-reactor/Dockerfile
Comment (kvaps) [inline modules/041-linstor/images/drbd-reactor/Dockerfile:1]: drbd-utils v9.24.0 just been released, let's update to stable version
https://lists.linbit.com/pipermail/drbd-announce/2023-June/000663.html?utm_content=253159763&utm_medium=social&utm_source=twitter&hss_channel=tw-56531410

File: modules/041-linstor/images/linstor-server/Dockerfile
Comment (kvaps) [inline modules/041-linstor/images/linstor-server/Dockerfile:1]: drbd-utils v9.24.0 just been released, let's update to stable version
https://lists.linbit.com/pipermail/drbd-announce/2023-June/000663.html?utm_content=253159763&utm_medium=social&utm_source=twitter&hss_channel=tw-56531410

**Review by konstantin-axenov — APPROVED** [review]

**Pull request merged**